windows/utils/win_get_vmtools_version_build.yml

```diff
@@ -3,20 +3,46 @@
 ---
 # Get installed VMware Tools version and build number in Windows guest OS
 # Return:
-#   vmtools_version: VMware Tools version
-#   vmtools_build_number: VMware Tools build number
+#   vmtools_version: VMware Tools version.
+#   vmtools_build_number: VMware Tools build number.
+#   vmtools_toolboxcmd_exists: Whether 'VMwareToolboxCmd.exe' exists in guest OS.
 #
 - name: "Initialize the facts of VMware Tools version and build info"
   ansible.builtin.set_fact:
     vmtools_info_from_vmtoolsd: ''
     vmtools_version: ''
     vmtools_major_version: ''
     vmtools_build_number: ''
+    vmtools_toolboxcmd_exists: false
+    vmtools_toolboxcmd_path: "C:\\Program Files\\VMware\\VMware Tools\\VMwareToolboxCmd.exe"
+    vmtoolsd_path: "C:\\Program Files\\VMware\\VMware Tools\\vmtoolsd.exe"
 
-- name: "Execute 'VMwareToolboxCmd.exe' to get VMware Tools info in guest OS"
+- name: "Check whether 'VMwareToolboxCmd.exe' exists"
+  include_tasks: ../utils/win_check_file_exist.yml
+  vars:
+    win_check_file_exist_file: "{{ vmtools_toolboxcmd_path }}"
+
+- name: "Set fact of 'VMwareToolboxCmd.exe' existence status"
+  ansible.builtin.set_fact:
+    vmtools_toolboxcmd_exists: "{{ win_check_file_exist_result }}"
+
+- name: "'VMwareToolboxCmd.exe' does not exist"
+  when: not vmtools_toolboxcmd_exists
+  block:
+    - name: "Check whether 'vmtoolsd.exe' exists"
+      include_tasks: ../utils/win_check_file_exist.yml
+      vars:
+        win_check_file_exist_file: "{{ vmtoolsd_path }}"
+    - name: "'vmtoolsd.exe' does not exist"
+      ansible.builtin.assert:
+        that:
+          - win_check_file_exist_result
+        fail_msg: "File '{{ vmtoolsd_path }}' does not exist in guest OS."
+
+- name: "Get VMware Tools info in guest OS"
   include_tasks: win_execute_cmd.yml
   vars:
-    win_powershell_cmd: "& 'C:\\Program Files\\VMware\\VMware Tools\\VMwareToolboxCmd.exe' -v"
+    win_powershell_cmd: "& '{{ vmtools_toolboxcmd_exists | ternary(vmtools_toolboxcmd_path, vmtoolsd_path) }}' -v"
```

Comment:
Since `vmtoolsd_path` must exist at here, why not directly use `{{ vmtoolsd_path }} -v`?

Reply:
Right, changed to use it directly. Thanks.